ui/widgets/calendar.js

```diff
@@ -133,12 +125,31 @@ return $.widget( "ui.calendar", {
 		this._addClass( $( event.currentTarget ), null, "ui-state-hover" );
 	},
 
+	_select: function( event ) {
+		var oldValue = this.options.value ? this.options.value.getTime() : "";
+
+		this._setOption( "value", new Date( $( event.currentTarget ).data( "timestamp" ) ) );
```

Comment:
You're suggesting to replace `new Date( $( event.currentTarget ).data( "timestamp" ) )` with `this.activeDescendant` right?

Don't think so as `this.activeDescendant` would be the current day, not the selected one.  `this.activeDescendant` will be updated in the next line, in `_updateDayElement`.